File: frontend/nordicwebhub-client/src/types/serviceOrder.ts
export type ServiceOrderStatus =
  | 'Pending'
  | 'Approved'
  | 'Paid'
  | 'Cancelled'

export type ServiceOrder = {
  id: number
  companyId: number
  companyName: string
  customerId: string
  customerEmail: string
  servicePackageId: number
  servicePackageName: string
  title: string
  notes: string
  amount: number
  status: ServiceOrderStatus
  paymentReference: string | null
  createdAt: string
  updatedAt: string
  paidAt: string | null
}

export type CreateServiceOrderDto = {
  servicePackageId: number
  title: string
  notes: string
  paymentReference?: string
}

export type UpdateServiceOrderStatusDto = {
  status: ServiceOrderStatus
}

File: backend/NordicWebHub.Api/Models/ServiceOrder.cs
using NordicWebHub.Api.Models.Enums;

namespace NordicWebHub.Api.Models;

public class ServiceOrder
{
    public int Id { get; set; }

    public int CompanyId { get; set; }

    public Company Company { get; set; } = null!;

    public string CustomerId { get; set; } = string.Empty;

    public ApplicationUser Customer { get; set; } = null!;

    public int ServicePackageId { get; set; }

    public ServicePackage ServicePackage { get; set; } = null!;

    public string Title { get; set; } = string.Empty;

    public string Notes { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    public ServiceOrderStatus Status { get; set; } = ServiceOrderStatus.Pending;

    public string? PaymentReference { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? PaidAt { get; set; }
}
